In [1]:
import pandas as pd
import numpy as np
import os

In [2]:
ls

 C 드라이브의 볼륨에는 이름이 없습니다.
 볼륨 일련 번호: E070-A7AC

 C:\Users\TFG5076XG\Documents\python\pandas 디렉터리

2021-11-26  오후 02:14    <DIR>          .
2021-11-26  오후 02:14    <DIR>          ..
2021-11-26  오후 02:14    <DIR>          .ipynb_checkpoints
2021-11-26  오후 01:53    <DIR>          output
2021-11-26  오후 01:58           437,385 pandas.ipynb
2021-11-26  오후 02:11            39,129 Pandas_퀴즈1.ipynb
2021-11-26  오후 02:14            66,960 Pandas_퀴즈2.ipynb
2021-11-24  오후 04:21         3,452,101 판다스.pdf
               4개 파일           3,995,575 바이트
               4개 디렉터리  440,367,603,712 바이트 남음


# Q1-1

```
아래의 파일을 이용하여 df_covid 데이타프레임을 생성하여라.
   
seoul-covid19_6_30_.csv    

```

In [1]:
import pandas as pd
import numpy as np
import os

In [4]:
ls data

 C 드라이브의 볼륨에는 이름이 없습니다.
 볼륨 일련 번호: A0F3-D988

 C:\workspace\WASSUP\WASSUP2\data_processing\data 디렉터리

2024-03-26  오후 11:30    <DIR>          .
2024-03-26  오후 11:29    <DIR>          ..
2024-03-19  오전 11:30           613,590 1_0.jpg
2024-03-19  오전 11:30           123,417 1_1.jpg
2024-03-19  오전 11:30           121,868 1_10.jpg
2024-03-19  오전 11:30            98,858 1_11.jpg
2024-03-19  오전 11:30           103,368 1_12.jpg
2024-03-19  오전 11:30           107,841 1_13.jpg
2024-03-19  오전 11:30            42,061 1_14.jpg
2024-03-19  오전 11:30           121,875 1_15.jpg
2024-03-19  오전 11:30           141,150 1_16.jpg
2024-03-19  오전 11:30           148,935 1_17.jpg
2024-03-19  오전 11:30            90,730 1_18.jpg
2024-03-19  오전 11:30            84,360 1_19.jpg
2024-03-19  오전 11:30           130,042 1_2.jpg
2024-03-19  오전 11:30            57,299 1_20.jpg
2024-03-19  오전 11:30           107,701 1_21.jpg
2024-03-19  오전 11:30           109,763 1_3.jpg
2024-03-19  오전 11:30            56,713 1_4.jpg
2024

In [5]:
df_covid = pd.read_csv('./data/seoul-covid19_6_30_.csv')
df_covid

연번     환자    확진일   거주지     여행력          접촉력         조치사항
0     1321  12830  6.30.  영등포구       -  영등포구 확진자 접촉            -
1     1320  12829  6.30.   관악구       -         확인 중        보라매병원
2     1319  12831  6.30.  영등포구       -         확인 중      중앙대학교병원
3     1318  12808  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
4     1317  12809  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
...    ...    ...    ...   ...     ...          ...          ...
1316     5      9  1.31.   성북구       -        #5 접촉    서울의료원(퇴원)
1317     4      7  1.30.   마포구  중국 우한시        해외 접촉    서울의료원(퇴원)
1318     3      6  1.30.   종로구       -        #3 접촉  서울대학교병원(퇴원)
1319     2      5  1.30.   중랑구  중국 우한시        해외 접촉    서울의료원(퇴원)
1320     1      2  1.24.   강서구  중국 우한시        해외 접촉  국립중앙의료원(퇴원)

[1321 rows x 7 columns]

# Q1-2
```
'연번' 컬럼은 삭제하고  '환자' 컬럼 값 기준으로 정렬한 후 '환자' 컬럼을 인덱스로 설정하여라. 
```

In [8]:
df_covid=df_covid.drop("연번",axis=1)

In [10]:
df_covid.head(10)

확진일  거주지     여행력    접촉력         조치사항
환자                                        
2   1.24.  강서구  중국 우한시  해외 접촉  국립중앙의료원(퇴원)
5   1.30.  중랑구  중국 우한시  해외 접촉    서울의료원(퇴원)
6   1.30.  종로구       -  #3 접촉  서울대학교병원(퇴원)
7   1.30.  마포구  중국 우한시  해외 접촉    서울의료원(퇴원)
9   1.31.  성북구       -  #5 접촉    서울의료원(퇴원)
10  1.31.  종로구       -  #6 접촉  서울대학교병원(퇴원)
11  1.31.  종로구       -  #6 접촉  서울대학교병원(퇴원)
13   2.2.  인천시   우한 교민  해외 접촉  국립중앙의료원(퇴원)
19   2.5.  송파구    싱가포르  해외 접촉    서울의료원(퇴원)
21   2.5.  성북구       -  #6 접촉  서울대학교병원(퇴원)

# Q1-3
어느 지역에서 제일 많은 확진자가 발생했는지 1~5위 까지 출력하여라 (거주지 기준)

In [11]:
df_group = df_covid.groupby('거주지').count().sort_values(by='확진일', ascending=False)
df_group

확진일  여행력  접촉력  조치사항
거주지                      
관악구   117  117  117   117
구로구    86   86   86    86
강남구    83   83   83    83
강서구    78   78   78    78
양천구    70   70   70    70
영등포구   65   65   65    65
도봉구    62   62   62    62
서초구    56   56   56    56
송파구    55   55   55    55
은평구    53   53   53    53
동작구    53   53   53    53
용산구    50   50   50    50
성동구    49   49   49    49
노원구    43   43   43    43
마포구    39   39   39    39
동대문구   38   38   38    38
강동구    35   35   35    35
서대문구   35   35   35    35
성북구    33   33   33    33
중랑구    33   33   33    33
금천구    32   32   32    32
강북구    24   24   24    24
종로구    22   22   22    22
광진구    17   17   17    17
중구     12   12   12    12
광명시     8    8    8     8
고양시     7    7    7     7
남양주     5    5    5     5
성남시     5    5    5     5
김포시     5    5    5     5
인천시     5    5    5     5
스페인     4    4    4     4
양평      3    3    3     3
부천시     3    3    3     3
양주시     3    3    3     3
수원시     3    3    3     3
의정부     2    2    2     2
용인시     2    2    2     2
구리시     2    2    2     2
대구시     2    2    2     2
천안시     2    2    2     2
대전시     2    2    2     2
재외국민    1    1    1     1
주소불명    1    1    1     1
파주시     1    1    1     1
의왕시     1    1    1     1
평택시     1    1    1     1
아산시     1    1    1     1
경북      1    1    1     1
양산시     1    1    1     1
안양시     1    1    1     1
안양      1    1    1     1
안산시     1    1    1     1
기타      1    1    1     1
시흥시     1    1    1     1
부평      1    1    1     1
미국      1    1    1     1
국외거주    1    1    1     1
군포시     1    1    1     1
화성시     1    1    1     1


# Q1-4
확진자가 가장 많이 발생한 1, 2, 3위 지역의 데이타만 출력하여라 

# Q1-5
'확진일' 컬럼값을 아래와 같은 형태로 변경하여라
<br>
'1.24.' => '2020-01-24'

# Q1-6
'확진일' 컬럼을 시계열 데이타로 형태를 변경하여라 

# Q1-7
3월에 발생한 확진자의 데이타만 출력하여라.

# Q1-8
주말에 발생한 확진자의 데이타만 출력하여라.

# Q1-9
'콜센터'와 관련된 확진자의 데이타를 출력하여라. 

## Q1-10
'우한'과 관련된 확진자의 데이타만 출력하여라. 

## Q1-11
사망자는 모두 몇명인가?

# Q2-1

아래의 파일을 이용하여 데이타프레임을 생성하여라.<br>
   
전국 평균 분양가격.csv 

# Q2-2

데이터프레임의 행과 열을 서로 변경하여라.

# Q2-3

1행을 삭제하고 2행을 컬럼으로 지정하여라. 

# Q2-4

'지역' 컬럼을 시계열 데이타로 변경하고 인덱스로 설정하여라. 

# Q2-5

2015년 6월 이후의 데이타만 출력하여라 

# Q2-6

2014년도의 서울, 인천 경기 컬럼만 출력하여라. 

# Q2-7
csv 파일 형태로 현재의 데이타프레임을 저장하여라. 
<br>
output/평균분양가격_서울인천경기.csv